app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ActiveRecord::Base
+  has_many :tweets
+  has_secure_password
+  validates :username, presence: true
+  validates :email, presence: true
+  validates :password, presence: true
+
+  def slug
+    self.username.gsub(/[[:punct:]]/, '').gsub(/\s/, '-')
+  end
+
+  def self.find_by_slug(slug)
+    User.all.find { |user| user.slug == slug }
```

Comment:
Unlike in the get '/users/:slug' action in the UsersController, here User.all.find is the appropriate method, since the `slug` attribute doesn't exist in the database.